cmd/alertmanager/main.go

```diff
@@ -427,3 +436,12 @@ func mustHostname() string {
 	}
 	return hostname
 }
+
+func md5HashAsMetricValue(data []byte) float64 {
+	sum := md5.Sum(data)
+	// We only want 48 bits to avoid NaN as a float64 only has a 53 bit mantissa.
+	smallSum := sum[0:6]
+	var bytes = make([]byte, 8)
```

Comment:
it panics with index out of bound when we don't have a byte slice with length >= 8, and the type of `sum` is `[16]byte`, which we can't pass directly into the function.